confluent_kafka/src/Consumer.c

```diff
@@ -743,6 +744,67 @@ static PyObject *Consumer_poll (Handle *self, PyObject *args,
 }
 
 
+static PyObject *Consumer_consume (Handle *self, PyObject *args,
+                                        PyObject *kwargs) {
+        size_t num_messages = 100;
+        double tmout = -1.0f;
+        static char *kws[] = { "timeout", "num_messages", NULL };
+        rd_kafka_message_t **rkmessages;
+        PyObject *msglist;
+        rd_kafka_queue_t *rkqu = NULL;
+        CallState cs;
+
+        if (!self->rk) {
+                PyErr_SetString(PyExc_RuntimeError,
+                                "Consumer closed");
+                return NULL;
+        }
+
+        if (!PyArg_ParseTupleAndKeywords(args, kwargs, "|nd", kws,
+					 (Py_ssize_t *)&num_messages, &tmout))
+		return NULL;
+
+        CallState_begin(self, &cs); // This unlocks GIL
+
+        rkqu = rd_kafka_queue_get_consumer(self->rk);
```

Comment:
Think this returns `NULL` if consumer was not properly initialized, which leads to a segfault further down; maybe just need to check this and exit if `NULL`?

Reply:
That should not happen, so no need to check